opal/corelib/string.rb

```diff
@@ -917,7 +918,8 @@ def scan(pattern, &block)
         if (block === nil) {
           match.length == 1 ? result.push(match[0]) : result.push(#{`match_data`.captures});
         } else {
-          match.length == 1 ? block(match[0]) : block.call(self, #{`match_data`.captures});
+          arrFunc = block.length == 1 ? 'call' : 'apply';
+          match.length == 1 ? block(match[0]) : block[arrFunc](self, #{`match_data`.captures});
```

Comment:
I think this can be rewritten using [`Opal.yieldX`](https://github.com/opal/opal/blob/9fd983b0190b4ef45e8da49bed23cce25a4f4b8d/opal/corelib/runtime.js#L1427)

also this

```
#{`match_data`.captures}
```

can be replaced with plain JS

```
match_data.captures
```